vyper/semantics/types/utils.py

```diff
@@ -127,12 +130,86 @@ def _failwith(type_name):
         # cases where the object in the namespace is an uninstantiated
         # type object, ex. Bytestring or DynArray (with no length provided).
         # call from_annotation to produce a better error message.
-        typ_.from_annotation(node)
+        typ_.from_annotation(node, namespace._constants)
 
     return typ_
 
 
-def get_index_value(node: vy_ast.Index) -> int:
+def prefold(node: vy_ast.VyperNode) -> Any:
```

Comment:
i feel like this function belongs in either `vyper/ast/` or something like `vyper/ast/pre_typecheck` to indicate that it occurs before regular semantic analysis